tests/foreman/api/test_docker.py

```diff
@@ -0,0 +1,662 @@
+
+    :return: A dictionary representing the created repository.
+
+    """
+
+    if name is None:
+        name = gen_string(gen_choice(STRING_TYPES), 15)
+    if upstream_name is None:
+        upstream_name = u'busybox'
+
+    return entities.Repository(
+        product=prod_id,
+        content_type=u'docker',
+        name=name,
+        docker_upstream_name=u'busybox',
+        url=DOCKER_REGISTRY_HUB
+    ).create()
+
+
+def _create_content_view(org_id, composite=False):
+    """Creates a contenr view.
+
+    :param id org_id: The ID for an existing organization.
+    :param bool composite: Whether the content view is composite or not.
+    Defaults to ``False``.
+
+    :return: A dictionary representing the created content view.
+
+    """
+    return entities.ContentView(
+        organization=org_id, composite=composite).create()
+
+
+def _add_repo_to_content_view(repo_id, cv_id):
+    """Adds a repository to an existing content view.
+
+    :param int repo_id: The ID for an existing repository.
+    :param int cv_id: The ID for an existing content view.
+
+    """
+    client.put(
+        entities.ContentView(id=cv_id).path(),
+        auth=get_server_credentials(),
+        verify=False,
+        data={u'repository_ids': [repo_id]}
+    ).raise_for_status()
+
+
+def _add_content_view_to_composite_view(cv_id, cv_version_id):
+    """Adds a published content view to a composite content view.
+
+    :param int cv_id: The ID for an existing composite content view.
+    :param int cv_version_id: The ID for a published non-composite
+    content view.
+
+    """
+    client.put(
+        entities.ContentView(id=cv_id).path(),
+        auth=get_server_credentials(),
+        verify=False,
+        data={u'component_ids': [cv_version_id]}
+    ).raise_for_status()
+
+
+def _publish_content_view(cv_id):
+    """Publishes an existing content view.
+
+    :param int cv_id: The ID for an existing content view.
+
+    :return: A boolean indicating whether the publication worked or not.
+
+    """
+    task_status = entities.ContentView(id=cv_id).publish()
+
+    return task_status['result'] == u'success'
+
+
+@ddt
+class DockerRepositoryTestCase(APITestCase):
+    """Tests specific to using ``Docker`` repositories."""
+
+    @classmethod
+    def setUpClass(cls):  # noqa
+        """Create an organization and product which can be re-used in tests."""
+        cls.org_id = entities.Organization().create()['id']
+
+    @run_only_on('sat')
+    @data(
+        gen_string('alpha', 15),
+        gen_string('alphanumeric', 15),
+        gen_string('numeric', 15),
+        gen_string('latin1', 15),
+        gen_string('utf8', 15),
+        gen_string('html', 15),
+    )
+    def test_create_one_docker_repo(self, name):
+        """@Test: Create one Docker-type repository
+
+        @Assert: A repository is created with a Docker image.
+
+        @Feature: Docker
+
+        """
+        upstream_name = u'busybox'
+        content_type = u'docker'
```

Comment:
I see two uses for `upstream_name`. But `content_type` is still only used once in this test.